PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:15 + add:0, change matches:
-    _inherit = "account.move"
-    _inherit = "account.move.line"
-    _inherit = "account.payment"
-    _inherit = "account.tax"
-    _inherit = 'account.analytic.applicability'
-    _inherit = 'account.analytic.account'
-    _inherit = 'hr.department'
-    _inherit = 'hr.employee'
-    _inherit = 'hr.employee.public'
-    _inherit = 'ir.actions.report'
-    _inherit = 'ir.attachment'
-    _inherit = "product.product"
-    _inherit = "product.template"
-    _inherit = "res.company"
-    _inherit = 'res.config.settings'

Total Changes: 40

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_expense/models/account_move.py
+++ b/addons/hr_expense/models/account_move.py
@@ -7,7 +7,7 @@ from odoo.tools.misc import frozendict
 
 
 class AccountMove(models.Model):
-    _inherit = "account.move"
+    _inherit = ["account.move"]
 
     expense_sheet_id = fields.Many2one(comodel_name='hr.expense.sheet', ondelete='set null', copy=False, index='btree_not_null')
     show_commercial_partner_warning = fields.Boolean(compute='_compute_show_commercial_partner_warning')

--- a/addons/hr_expense/models/account_move_line.py
+++ b/addons/hr_expense/models/account_move_line.py
@@ -7,7 +7,7 @@ from odoo.tools.misc import frozendict
 
 
 class AccountMoveLine(models.Model):
-    _inherit = "account.move.line"
+    _inherit = ["account.move.line"]
 
     expense_id = fields.Many2one('hr.expense', string='Expense', copy=True) # copy=True, else we don't know price is tax incl.
 

--- a/addons/hr_expense/models/account_payment.py
+++ b/addons/hr_expense/models/account_payment.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class AccountPayment(models.Model):
-    _inherit = "account.payment"
+    _inherit = ["account.payment"]
 
     expense_sheet_id = fields.Many2one(related='move_id.expense_sheet_id')
 

--- a/addons/hr_expense/models/account_tax.py
+++ b/addons/hr_expense/models/account_tax.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class AccountTax(models.Model):
-    _inherit = "account.tax"
+    _inherit = ["account.tax"]
 
     def _hook_compute_is_used(self, taxes_to_compute):
         # OVERRIDE in order to fetch taxes used in expenses

--- a/addons/hr_expense/models/analytic.py
+++ b/addons/hr_expense/models/analytic.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class AccountAnalyticApplicability(models.Model):
-    _inherit = 'account.analytic.applicability'
+    _inherit = ['account.analytic.applicability']
     _description = "Analytic Plan's Applicabilities"
 
     business_domain = fields.Selection(
@@ -25,7 +25,7 @@ class AccountAnalyticApplicability(models.Model):
 
 
 class AccountAnalyticAccount(models.Model):
-    _inherit = 'account.analytic.account'
+    _inherit = ['account.analytic.account']
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_account_in_analytic_distribution(self):

--- a/addons/hr_expense/models/hr_department.py
+++ b/addons/hr_expense/models/hr_department.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class HrDepartment(models.Model):
-    _inherit = 'hr.department'
+    _inherit = ['hr.department']
 
     def _compute_expense_sheets_to_approve(self):
         expense_sheet_data = self.env['hr.expense.sheet']._read_group([('department_id', 'in', self.ids), ('state', '=', 'submit')], ['department_id'], ['__count'])

--- a/addons/hr_expense/models/hr_employee.py
+++ b/addons/hr_expense/models/hr_employee.py
@@ -3,8 +3,8 @@
 from odoo import fields, models, api
 
 
-class EmployeeBase(models.AbstractModel):
-    _inherit = 'hr.employee.base'
+class HrEmployeeBase(models.AbstractModel):
+    _inherit = ['hr.employee.base']
 
     filter_for_expense = fields.Boolean(store=False, search='_search_filter_for_expense', groups="hr.group_hr_user")
 
@@ -30,8 +30,8 @@ class EmployeeBase(models.AbstractModel):
         return res
 
 
-class Employee(models.Model):
-    _inherit = 'hr.employee'
+class HrEmployee(models.Model):
+    _inherit = ['hr.employee']
 
     def _group_hr_expense_user_domain(self):
         # We return the domain only if the group exists for the following reason:
@@ -67,8 +67,8 @@ class Employee(models.Model):
         return super()._get_user_m2o_to_empty_on_archived_employees() + ['expense_manager_id']
 
 
-class EmployeePublic(models.Model):
-    _inherit = 'hr.employee.public'
+class HrEmployeePublic(models.Model):
+    _inherit = ['hr.employee.public']
 
     expense_manager_id = fields.Many2one('res.users', readonly=True)
 

--- a/addons/hr_expense/models/hr_expense.py
+++ b/addons/hr_expense/models/hr_expense.py
@@ -10,7 +10,6 @@ from odoo.tools import email_split, float_repr, float_round, is_html_empty
 
 
 class HrExpense(models.Model):
-    _name = "hr.expense"
     _inherit = ['mail.thread.main.attachment', 'mail.activity.mixin', 'analytic.mixin']
     _description = "Expense"
     _order = "date desc, id desc"

--- a/addons/hr_expense/models/hr_expense_sheet.py
+++ b/addons/hr_expense/models/hr_expense_sheet.py
@@ -26,7 +26,6 @@ class HrExpenseSheet(models.Model):
                     Manager                 Always
         =================================================================================
     """
-    _name = "hr.expense.sheet"
     _inherit = ['mail.thread.main.attachment', 'mail.activity.mixin']
     _description = "Expense Report"
     _order = "accounting_date desc, id desc"

--- a/addons/hr_expense/models/ir_actions_report.py
+++ b/addons/hr_expense/models/ir_actions_report.py
@@ -5,7 +5,7 @@ from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
 
 
 class IrActionsReport(models.Model):
-    _inherit = 'ir.actions.report'
+    _inherit = ['ir.actions.report']
 
     def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         # OVERRIDE

--- a/addons/hr_expense/models/ir_attachment.py
+++ b/addons/hr_expense/models/ir_attachment.py
@@ -2,7 +2,7 @@ from odoo import models, api
 
 
 class IrAttachment(models.Model):
-    _inherit = 'ir.attachment'
+    _inherit = ['ir.attachment']
 
     @api.model_create_multi
     def create(self, vals_list):

--- a/addons/hr_expense/models/product_product.py
+++ b/addons/hr_expense/models/product_product.py
@@ -2,7 +2,7 @@ from odoo import api, fields, models, _
 
 
 class ProductProduct(models.Model):
-    _inherit = "product.product"
+    _inherit = ["product.product"]
 
     standard_price_update_warning = fields.Char(compute="_compute_standard_price_update_warning")
 

--- a/addons/hr_expense/models/product_template.py
+++ b/addons/hr_expense/models/product_template.py
@@ -6,7 +6,7 @@ from odoo.tools.sql import column_exists, create_column
 
 
 class ProductTemplate(models.Model):
-    _inherit = "product.template"
+    _inherit = ["product.template"]
 
     @api.model
     def default_get(self, fields):

--- a/addons/hr_expense/models/res_company.py
+++ b/addons/hr_expense/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = "res.company"
+    _inherit = ["res.company"]
 
     expense_journal_id = fields.Many2one(
         "account.journal",

--- a/addons/hr_expense/models/res_config_settings.py
+++ b/addons/hr_expense/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     hr_expense_alias_prefix = fields.Char(
         'Default Alias Name for Expenses',
